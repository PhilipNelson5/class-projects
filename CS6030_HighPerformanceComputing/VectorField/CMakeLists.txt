cmake_minimum_required(VERSION 3.14)

project(
    FinalProject
    LANGUAGES CXX
)

set(CMAKE_CXX_STANDARD 17 CACHE INTERNAL "")
set(CMAKE_CXX_STANDARD_REQUIRED ON CACHE INTERNAL "")
set(CMAKE_CXX_EXTENSIONS OFF CACHE INTERNAL "")
# add_compile_options(-Wall -Wextra -Wpedantic -Werror)

set (MAKE_TESTS FALSE)
if (${MAKE_TESTS})
    enable_testing()
endif()

add_subdirectory(Expected)
add_subdirectory(Math)
add_subdirectory(VectorField)
add_subdirectory(Graphics)
add_subdirectory(Helpers)
add_subdirectory(driver_cuda)
add_subdirectory(driver_mpi)
add_subdirectory(driver_serial)
add_subdirectory(driver_thread)
