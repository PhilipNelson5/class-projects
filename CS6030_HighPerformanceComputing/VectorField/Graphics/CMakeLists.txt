set(TARGET Graphics)

add_library(${TARGET} INTERFACE
    src/Color/Color.hpp
    src/Draw/Arrow.hpp
    src/Draw/Circle.hpp
    src/Draw/Curve.hpp
    src/Draw/Line.hpp
    src/Image/Image.hpp
    src/Image/Pixel.hpp
)

target_link_libraries(${TARGET} INTERFACE Math)

target_include_directories(${TARGET} INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/src")