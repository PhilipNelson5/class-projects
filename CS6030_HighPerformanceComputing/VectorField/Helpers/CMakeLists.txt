set(TARGET Helpers)

add_library(${TARGET} INTERFACE
    src/Helpers/AssertHelper.hpp
    src/Helpers/CudaHelpers.hpp
    src/Helpers/VectorFieldImageHelpers.hpp
)

target_include_directories(${TARGET} INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/src")