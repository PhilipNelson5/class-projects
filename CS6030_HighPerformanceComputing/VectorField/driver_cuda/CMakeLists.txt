include(CheckLanguage)

check_language(CUDA)

if (CMAKE_CUDA_COMPILER)

    message(STATUS "CUDA AVAILIABLE")
	  enable_language(CUDA)

	  if(NOT DEFINED CMAKE_CUDA_STANDARD)
		    set(CMAKE_CUDA_STANDARD 14)
		    set(CMAKE_CUDA_STANDARD_REQUIRED ON)
	  endif()

    include(FetchContent)
    FetchContent_Declare(
        cxxopts
        GIT_REPOSITORY https://github.com/jarro2783/cxxopts.git
        GIT_TAG        v3.0.0
    )
    FetchContent_MakeAvailable(cxxopts)

	  set(TARGET run_cuda)
	  add_executable(${TARGET} main.cu)
	  set_property(TARGET ${TARGET} PROPERTY CUDA_ARCHITECTURES OFF)
	  target_link_libraries(${TARGET} PUBLIC cxxopts::cxxopts VectorField Graphics Helpers)

else()

    message(STATUS "CUDA UNAVAILIABLE")

endif()
