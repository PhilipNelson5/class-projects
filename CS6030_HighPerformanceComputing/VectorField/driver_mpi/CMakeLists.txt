find_package(MPI REQUIRED)

include(FetchContent)
FetchContent_Declare(
    cxxopts
    GIT_REPOSITORY https://github.com/jarro2783/cxxopts.git
    GIT_TAG        v3.0.0
)
FetchContent_MakeAvailable(cxxopts)


set(TARGET run_mpi)
add_executable(${TARGET} main.cpp)
target_link_libraries(${TARGET} PUBLIC cxxopts::cxxopts MPI::MPI_CXX VectorField Graphics Helpers)

set(TARGET_GRID run_mpi_grid)
add_executable(${TARGET_GRID} main_grid.cpp)
target_link_libraries(${TARGET_GRID} PUBLIC cxxopts::cxxopts MPI::MPI_CXX VectorField Graphics Helpers)
