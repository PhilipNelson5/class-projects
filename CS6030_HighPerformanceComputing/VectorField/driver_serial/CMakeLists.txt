set(TARGET run_serial)
add_executable(${TARGET} main.cpp)

include(FetchContent)
FetchContent_Declare(
    cxxopts
    GIT_REPOSITORY https://github.com/jarro2783/cxxopts.git
    GIT_TAG        v3.0.0
)
FetchContent_MakeAvailable(cxxopts)

target_link_libraries(${TARGET} PUBLIC cxxopts::cxxopts VectorField Graphics Helpers)