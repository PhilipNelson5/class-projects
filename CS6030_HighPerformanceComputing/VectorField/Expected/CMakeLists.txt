set(TARGET Expected)

add_library(${TARGET} INTERFACE
    src/Expected/Expected.hpp
)

target_include_directories(${TARGET} INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/src")

if (${MAKE_TESTS})
    add_subdirectory(test)
endif()
